plugins/sfThumbnailPlugin/lib/sfImageMagickAdapter.class.php

```diff
@@ -217,6 +217,25 @@ public function loadFile($thumbnail, $image)
     return true;
   }
 
+  /**
+   * Determine file mime type using the PHP fileinfo library.
+   *
+   * @param string  file we want the mime type for
+   *
+   * @return string  Mime type
+   */
+  public static function getMimeType($file)
+  {
+    // Use fileinfo to figure out file mimetype.
+    $file_info = new finfo(FILEINFO_MIME);
+    $mimeInfo = $file_info->buffer(file_get_contents($file));
```

Comment:
Wouldn't be more efficient to use `finfo::file` or `finfo_file` so we don't have to load the whole file into a string?